### PR TITLE
[GLUTEN-3252][CH] fix race condition at the global variable of isAdaptiveContext in ColumnarOverrideRules

### PR DESCRIPTION
**Pull request by zheniantoushipashi**
## What changes were proposed in this pull request?

[GLUTEN-3252][CH] fix race condition at the global variable of isAdaptiveContext in ColumnarOverrideRules

## How was this patch tested?

test  GlutenClickHouseTPCHParquetAQEConcurrentSuite  can  reproduce

### CODE REVIEW

**Comment by ulysses-you** [comment]
good catch!

File: gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala [hunk 1]
```diff
@@ -731,8 +731,6 @@ case class ColumnarOverrideRules(session: SparkSession)
   private lazy val transformPlanLogLevel = GlutenConfig.getConf.transformPlanLogLevel
   @transient private lazy val planChangeLogger = new PlanChangeLogger[SparkPlan]()
 
-  // Tracks whether the columnar rule is called through AQE.
-  private var isAdaptiveContext: Boolean = false
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:735]: Thanks so much for your fix!
Can we just use `ThreadLocal` type for `isAdaptiveContext` to fix this issue?

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:735]: `localProperties` is also a `ThreadLocal` parameter, using `localProperties` is OK to me.

Comment (baibaichen) [reply gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala:735]: @PHILO-HE 
I think `LocalProperty `is more suitable here, and there will be some similar variables that need to be added in the future.

**Review by PHILO-HE — APPROVED** [review]

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**